### PR TITLE
Feature/pelagos 4719 create rg info report

### PR DESCRIPTION
**Pull request by mickel1138**

### CODE REVIEW

File: src/Controller/UI/ReportResearchGroupDatasetStatusController.php [hunk 1]
```diff
@@ -10,6 +10,11 @@
 
 use App\Handler\EntityHandler;
 use Doctrine\Common\Collections\Collection;
+use Doctrine\DBAL\ForwardCompatibility\Result;
+use JMS\Serializer\SerializationContext;
+use JMS\Serializer\SerializerBuilder;
+use JMS\Serializer\SerializerInterface;
```
Comment (praneethpr) [inline src/Controller/UI/ReportResearchGroupDatasetStatusController.php:16]: not used

File: src/Controller/UI/ReportResearchGroupDatasetStatusController.php [hunk 2]
```diff
@@ -202,6 +206,112 @@ public function getGrpResearchGroupReport(Request $request, EntityHandler $entit
...
+     * @param string|null   $id              Research group id.
+     *
+     * @Route(
+     *     "/report-researchgroup/grp-detail-report/{id}",
+     *     name="pelagos_app_ui_reportresearchgroup_detailreport",
+     *     )
+     *
+     * @return Response|StreamedResponse A Symfony Response instance.
+     */
+    public function getGrpResearchDetailReport(Request $request, EntityHandler $entityHandler, string $id = null)
```
Comment (praneethpr) [inline src/Controller/UI/ReportResearchGroupDatasetStatusController.php:223]: Can add return type

Comment (praneethpr) [inline src/Controller/UI/ReportResearchGroupDatasetStatusController.php:223]: Can change the name of the function, seems like it is for GRP

File: src/Controller/UI/ReportResearchGroupDatasetStatusController.php [hunk 3]
```diff
@@ -202,6 +206,112 @@ public function getGrpResearchGroupReport(Request $request, EntityHandler $entit
...
+                'form' => $form->createView(),
+                'reportTitle' => 'Research Group Detail Report',
+            )
+        );
+    }
+
+    /**
+     * Get Research Group data as JSON.
+     *
+     * @param integer $researchGroupId
```
Comment (praneethpr) [inline src/Controller/UI/ReportResearchGroupDatasetStatusController.php:268]: Missing param description

**Review by praneethpr — APPROVED** [review]

File: src/Controller/UI/ReportResearchGroupDatasetStatusController.php [hunk 4]
```diff
@@ -202,6 +202,112 @@ public function getGrpResearchGroupReport(Request $request, EntityHandler $entit
...
 
+    /**
+     * Research group dataset status report for dataset monitoring.
+     *
+     * @param Request       $request         Message information for this Request.
+     * @param EntityHandler $entityHandler   The entity handler.
+     * @param string|null   $id              Research group id.
+     *
+     * @Route(
+     *     "/report-researchgroup/grp-detail-report/{id}",
```
Comment (fightingtexasaggie) [inline src/Controller/UI/ReportResearchGroupDatasetStatusController.php:213]: maybe 'grp' is a bit confusing in this route name. This is minor as users aren't really going to see it anyway.

**Review by fightingtexasaggie — COMMENTED** [review]
I think we need a link on the admin menu for this report.

**Pull request merged**